lib/src/chunking.rs

```diff
@@ -469,4 +799,132 @@ mod test {
+            vec![7, 30, 1],
+        ]
+        .iter()
+        .map(|data| ObjectSourceMetaSized {
+            meta: ObjectSourceMeta {
+                identifier: RcStr::from(format!("pkg{}.0", data[0])),
+                name: RcStr::from(format!("pkg{}", data[0])),
+                srcid: RcStr::from(format!("srcpkg{}", data[0])),
+                change_time_offset: 0,
+                change_frequency: data[1],
+            },
+            size: data[2] as u64,
+        })
+        .collect();
+
+        let packing = basic_packing(
+            &contentmeta_v0.as_slice(),
+            NonZeroU32::new(6).unwrap(),
+            None,
+        );
+        let structure: Vec<Vec<&str>> = packing
+            .iter()
+            .map(|bin| bin.iter().map(|pkg| &*pkg.meta.name).collect())
+            .collect();
+        let mut v0_expected_structure = vec![
+            vec![String::from("pkg3")],
+            vec![String::from("pkg4")],
+            vec![
+                String::from("pkg6"),
+                String::from("pkg7"),
+                String::from("pkg11"),
+            ],
+            vec![
+                String::from("pkg9"),
+                String::from("pkg8"),
+                String::from("pkg10"),
+            ],
+            vec![String::from("pkg1"), String::from("pkg2")],
+            vec![],
+        ];
+        assert_eq!(structure, v0_expected_structure);
+
+        //Derived
+
+        let mut contentmeta_v1: Vec<ObjectSourceMetaSized> = contentmeta_v0;
+        //Upgrade pkg1.0 to 1.1
+        contentmeta_v1[0].meta.identifier = RcStr::from("pkg1.1");
+        //Remove pkg7
+        contentmeta_v1.remove(contentmeta_v1.len() - 1);
+        //Add pkg12
+        contentmeta_v1.push(ObjectSourceMetaSized {
+            meta: ObjectSourceMeta {
+                identifier: RcStr::from("pkg5.0"),
+                name: RcStr::from("pkg5"),
+                srcid: RcStr::from("srcpkg5"),
+                change_time_offset: 0,
+                change_frequency: 42,
+            },
+            size: 100000,
+        });
+
+        let mut metadata_with_ostree_commit = vec![vec![String::from("ostree_commit")]];
+        metadata_with_ostree_commit.append(&mut v0_expected_structure);
+
+        let config = oci_spec::image::DescriptorBuilder::default()
+            .media_type(oci_spec::image::MediaType::ImageConfig)
+            .size(7023)
+            .digest("sha256:imageconfig")
+            .build()
+            .expect("build config descriptor");
+
+        let layers: Vec<oci_spec::image::Descriptor> = metadata_with_ostree_commit
+            .iter()
+            .map(|l| {
+                oci_spec::image::DescriptorBuilder::default()
+                    .media_type(oci_spec::image::MediaType::ImageLayerGzip)
+                    .size(100)
+                    .digest(format!("sha256:{}", l.len()))
+                    .annotations(HashMap::from([("Content".to_string(), l.join(","))]))
```

Comment:
Let's make this `"Content"` a `const CONTENT_ANNOTATION: &str = "Content";` above?